midterm-2023-11-06/code/5163-mihaela-milosevska/script.ts

```diff
@@ -0,0 +1,82 @@
+interface album{
+    name: string;
+    year; number;
+}
+
+interface band {
+    id: number;
+    name: string;
+    formed: number;
+    location: string;
+    genre: string;
+    members: string;
+    albums: number;
+}
+type BandSorter = (first: Band, second: Band) => number;
+
+async function getBands(): Promise<Band[]> {
+    const url = "https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-06/data/bands.json";
+    const response = await axios.get(url);
+    return response.data.metalBands;
+}
+
+function formatBand(band: Band): Record<string, unknown> {
+    const members = band.members.sort().slice(0, 5).join(', ');
+    const lastMember = band.members.length > 5 ? '...' : '';
+    const membersString = band.members.length < 3 ? members : `${members}, ${lastMember}`;
+    const [city, country] = band.location.split(', ');
+    return {
+        id: band.id,
+        name: band.name,
+        formed: band.formed,
+        location: city,
+        genre: band.genre,
+        members: membersString,
+        albums: band.albums.length,
+    };
+}
+
+async function main() {
+    const bands = await getBands();
+    const bandInfos = bands.map(formatBand);
+    let sortedBy = '';
+    let ascending = true;
+    const table = document.createElement('table');

```

Comment:
+10 for an alternative approach to the grid by using a `table` tag. However, using a `table` tag is very much frowned upon in modern web design, and with good cause.